src/zabapgit_objects.prog.abap

```diff
@@ -155,6 +171,198 @@ CLASS lcl_objects_activation IMPLEMENTATION.
 
   ENDMETHOD.                    "activate
 
+
+  METHOD use_new_activation_logic.
+
+    IF zcl_abapgit_persist_settings=>get_instance( )->read( )->get_experimental_features( ) = abap_true.
+
+      CALL FUNCTION 'FUNCTION_EXISTS'
+        EXPORTING
+          funcname           = 'DD_MASS_ACT_C3'    " Name of Function Module
+        EXCEPTIONS
+          function_not_exist = 1
+          OTHERS             = 2.
+
+      IF sy-subrc = 0.
+        rv_use_new_activation_logic = abap_true.
+      ENDIF.
+
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD activate_new.
+
+    DATA: lt_ddic     TYPE act_work_area,
+          lt_non_ddic TYPE act_work_area.
+
+    IF gt_objects IS INITIAL.
+      RETURN.
+    ENDIF.
+
+    zcl_abapgit_progress=>show( iv_key     = 'Activate'
+                                iv_current = '100'
+                                iv_total   = '100'
+                                iv_text    = '...' ).
+
+    filter_ddic_objects(
+      IMPORTING
+        et_ddic     = lt_ddic
+        et_non_ddic = lt_non_ddic ).
+
+    IF lines( lt_ddic ) > 0 AND iv_ddic = abap_true.
+
+      activate_ddic( lt_ddic ).
+
+    ENDIF.
+
+    IF lines( lt_non_ddic ) > 0.
+
+      activate_non_ddic( lt_non_ddic ).
+
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD activate_old.
+
+    IF gt_objects IS NOT INITIAL.
+
+      CALL FUNCTION 'RS_WORKING_OBJECTS_ACTIVATE'
+        EXPORTING
+          activate_ddic_objects  = iv_ddic
+          with_popup             = abap_true
+        TABLES
+          objects                = gt_objects
+        EXCEPTIONS
+          excecution_error       = 1
+          cancelled              = 2
+          insert_into_corr_error = 3
+          OTHERS                 = 4.
+      IF sy-subrc <> 0.
+        zcx_abapgit_exception=>raise( 'error from RS_WORKING_OBJECTS_ACTIVATE' ).
+      ENDIF.
+
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD filter_ddic_objects.
```

Comment:
do we have some objects in abapGit that identifies themselves as DDIC, but are not really DDIC? Or is there a difference between the function modules?

Reply:
Oh, I see. We have already the distinction between DDIC and non-DDIC, so filter_ddic_objects is superfluous. Will change the code.